ods_ci/tests/Tests/1300__model_registry/1302_model_registry_model_serving.robot

```diff
@@ -0,0 +1,277 @@
+    Data Connection Should Be Listed    name=${DC_S3_NAME}    type=${DC_S3_TYPE}    connected_workbench=${workbenches}
+    Open Data Science Project Details Page       project_title=${prj_title}    tab_id=workbenches
+    Wait Until Workbench Is Started     workbench_title=registry-wb
+    Upload File In The Workbench     filepath=${SAMPLE_ONNX_MODEL}    workbench_title=${WORKBENCH_TITLE}
+    ...         workbench_namespace=${PRJ_TITLE}
+    Upload File In The Workbench     filepath=${JUPYTER_NOTEBOOK_FILEPATH}    workbench_title=${WORKBENCH_TITLE}
+    ...         workbench_namespace=${PRJ_TITLE}
+    Launch And Access Workbench    workbench_title=${WORKBENCH_TITLE}
+    ...    username=${TEST_USER.USERNAME}     password=${TEST_USER.PASSWORD}
+    ...    auth_type=${TEST_USER.AUTH_TYPE}
+    Upload Certificate To Jupyter Notebook    ${CERTS_DIRECTORY}/domain.crt
+    Jupyter Notebook Should Run Successfully     ${JUPYTER_NOTEBOOK}
+
+
+*** Keywords ***
+Prepare Model Registry Test Setup
+    [Documentation]    Suite setup steps for testing Model Registry.
+    Set Library Search Order    SeleniumLibrary
+    RHOSi Setup
+    Launch Dashboard    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    ...    ${ODH_DASHBOARD_URL}    ${BROWSER.NAME}    ${BROWSER.OPTIONS}
+    Open Data Science Projects Home Page
+    Create Data Science Project    title=${PRJ_TITLE}    description=${PRJ_DESCRIPTION}
+    Create Project
+    Apply ServiceMeshMember Configuration
+    Run OC Commands And Capture Output
+    Run Update Notebook Script
+    Copy File    ${EXAMPLE_ISTIO_ENV}    ${ISTIO_ENV}
+    Append Key Value To Env File    ${ISTIO_ENV}    DOMAIN    ${DOMAIN}
+    File Should Exist    ${ISTIO_ENV}
+    Log File Content    ${ISTIO_ENV}
+    Generate ModelRegistry Certificates
+    Apply Db Config Samples    namespace=${NAMESPACE_MODEL-REGISTRY}
+    Create Model Registry Secrets
+    Fetch CA Certificate If RHODS Is Self-Managed
+
+Teardown Model Registry Test Setup
+    [Documentation]  Teardown Model Registry Suite
+    JupyterLibrary.Close All Browsers
+    ${return_code}    ${output}=    Run And Return Rc And Output
+    ...    oc delete project ${PRJ_TITLE} --force --grace-period=0
+    Should Be Equal As Integers	  ${return_code}	 0
+    Log    ${output}
+    RHOSi Teardown
+
+Create Project
+    [Documentation]    Create a project using oc command.
+    Create OpenShift Project    ${NAMESPACE_MODEL-REGISTRY}
+
+Run OC Commands And Capture Output
+    [Documentation]  Logs the Domain and Token capture.
+    ${domain}=    Get Domain
+    ${token}=    Get Token
+    Set Suite Variable    ${DOMAIN}    ${domain}
+    Set Suite Variable    ${TOKEN}    ${token}
+    Log    Domain: ${DOMAIN}
+    Log    Token: ${TOKEN}
+
+Generate ModelRegistry Certificates
+    [Documentation]    Generates OpenSSL certificates for Model-Registry using the generate_certs.sh script.
```

Comment:
This is strange to me, why do you need to generate all these certificates?
I would assume that RHOAI/ODH takes care of creating what is needed for Model Registry? Or would a end user be expected to create all of this by themselves?

Reply:
That is one of the use cases. The end user might be creating their own certificates.